driver/main_args.ml

```diff
@@ -731,15 +731,32 @@ let mk_extension f =
     Clflags.Extension.(List.map to_string all)
   in
   "-extension", Arg.Symbol (available_extensions, f),
-  "<extension>  Enable the extension (may be specified more than once)"
+  "<extension>  Enable the specified extension (may be specified more than once)"
+;;
+
+let mk_no_extension f =
+  let available_extensions =
+    Clflags.Extension.(List.map to_string all)
+  in
+  "-no-extension", Arg.Symbol (available_extensions, f),
+  "<extension>  Disable the specified extension (may be specified more than once)"
 ;;
 
 let mk_disable_all_extensions f =
   "-disable-all-extensions", Arg.Unit f,
-  " Disable all extensions, wherever they are specified; this flag\n\
-  \    overrides the -extension flag (whether specified before or after this\n\
-  \    flag), disables any extensions that are enabled by default, and\n\
-  \    ignores any extensions requested in OCAMLPARAM."
+  " Disable all extensions, wherever they have been specified; this\n\
+  \    flag overrides prior uses of the -extension flag, disables any\n\
+  \    extensions that are enabled by default, and causes future uses of\n\
+  \    the -extension or -no-extension flag to raise."
```

Comment:
```suggestion
  \    the -extension or -no-extension flag to error."
```

Why `-no-extension`? I think `-no-extension` should be allowed even when extensions are disabled.

Reply:
I can see this both ways.  Right now, this directly calls `disable`, which also fails, and I'm more inclined to believe _that_'s the right behavior.

Reply:
Nah, I'm convinced.